netty/src/main/java/io/grpc/netty/WriteQueue.java

```diff
@@ -126,20 +119,20 @@ ChannelFuture enqueue(QueuedCommand command, ChannelPromise promise, boolean flu
    */
   private void flush() {
     try {
-      boolean flushed = false;
-      while (queue.drainTo(writeChunk, DEQUE_CHUNK_SIZE) > 0) {
-        while (writeChunk.size() > 0) {
-          QueuedCommand cmd = writeChunk.poll();
-          channel.write(cmd, cmd.promise());
+      QueuedCommand cmd;
+      int i = 0;
+      while ((cmd = queue.poll()) != null) {
+        channel.write(cmd, cmd.promise());
+        if (++i == DEQUE_CHUNK_SIZE) {
+          i = 0;
+          // Flush each chunk so we are releasing buffers periodically. In theory this loop
+          // might never end as new events are continuously added to the queue, if we never
+          // flushed in that case we would be guaranteed to OOM.
+          channel.flush();
         }
-        // Flush each chunk so we are releasing buffers periodically. In theory this loop
-        // might never end as new events are continuously added to the queue, if we never
-        // flushed in that case we would be guaranteed to OOM.
-        flushed = true;
-        channel.flush();
       }
-      if (!flushed) {
-        // Must flush at least once
+      // Must flush at least once
+      if (i != 0) {
```

Comment:
I determined that flushing at least once is in order for `scheduleFlush` to function.